.github/workflows/rc.yml

```diff
@@ -70,15 +70,269 @@ jobs:
       - name: Audit
         run: |
           dev/release/run_rat.sh "${TAR_GZ}"
-      - uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b # v4.5.0
+      - name: Upload source archive
+        uses: actions/upload-artifact@65c4c4a1ddee5b72f698fdd19549f0f0fb45cf08 # v4.6.0
         with:
-          name: archive
+          name: release-source
           path: |
             apache-arrow-java-*
+  jni-ubuntu:
+    name: JNI ${{ matrix.platform.runs_on }} ${{ matrix.platform.arch }}
+    runs-on: ${{ matrix.platform.runs_on }}
+    needs:
+      - source
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - runs_on: ubuntu-latest
+            arch: "x86_64"
+            archery_arch: "amd64"
+    env:
+      # architecture name used for archery build
+      ARCH: ${{ matrix.platform.archery_arch }}
+      DOCKER_VOLUME_PREFIX: .docker/
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Download source archive
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: release-source
+      - name: Extract Download the latest Apache Arrow C++
+        run: |
+          tar -xf apache-arrow-java-*.tar.gz --strip-components=1
+      - name: Download the latest Apache Arrow C++
+        run: |
+          ci/scripts/download_cpp.sh
+      - name: Checkout apache/arrow-testing
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: apache/arrow-testing
+          path: arrow/testing
+      - name: Checkout apache/parquet-testing
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          repository: apache/parquet-testing
+          path: arrow/cpp/submodules/parquet-testing
```

Comment:
Do we need testing data? (I don't think we need to run C++ tests here?)

Reply:
OK. I'll disable C++ tests.
(I thought that we need to run C++ tests because it seems that we don't have enough test for JNI related codes.)

Reply:
The C++ tests we run here are just the same tests that apache/arrow already runs right? (I agree we need more testing for the JNI code itself, though...)

Reply:
Right but we use different build options that aren't covered by apache/arrow.

Reply:
Hmm if they're that different then should we be testing the config upstream? Or otherwise in that case I don't mind running the C++ tests here if our config is very different (it mostly just adds a lot of build time)

Reply:
OK. Let's do it in upstream.